src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -1759,6 +1763,12 @@ private Symbol GetDeclaredMember(NamespaceOrTypeSymbol container, TextSpan decla
 
             // look for any member with same declaration location
             var collection = name != null ? container.GetMembers(name) : container.GetMembersUnordered();
+            if (isKnownNamespace && name != null && collection is [NamespaceSymbol knownNamespace])
```

Comment:
>name != null [](http://example.com/codeflow?start=36&length=12)

Let's add an assert on entry into the function that ```name``` is not null if ```declarationSpan``` is true.

Reply:
➡️ Added this assertion. Will review what to do if tests fail.